lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextManager.cs

```diff
@@ -349,14 +351,33 @@ private void StartTask(StartTaskProto startTaskProto)
         private void HandleTaskException(TaskClientCodeException e)
         {
             LOGGER.Log(Level.Error, "TaskClientCodeException", e);
-            byte[] exception = ByteUtilities.StringToByteArrays(e.ToString());
-            TaskStatusProto taskStatus = new TaskStatusProto()
+            byte[] error;
+            try
+            {
+                error = ByteUtilities.SerializeToBinaryFormat(e);
+            }
+            catch (SerializationException se)
+            {
+                error = ByteUtilities.SerializeToBinaryFormat(
```

Comment:
As TaskClientCodeException is serializable, is it still possible to have SerializationException here?